HerPortal.ManagementShell/Program.cs

```diff
@@ -0,0 +1,228 @@
+
+            bool userConfirmation;
+            switch (devAction)
+            {
+                case "remove-user":
+                    adminAction.TryRemoveUser(foundUserOrDefault);
+                    break;
+                case "remove-las":
+                    userConfirmation = adminAction.ConfirmCustodianCodes(custodianCodes, userEmailAddress);
+                    if (userConfirmation)
+                    {
+                        adminAction.RemoveLas(custodianCodes, foundUserOrDefault);
+                    }
+
+                    break;
+                case "add-las":
+                    adminAction.DisplayUserStatus(userStatus);
+
+
+                        userConfirmation = adminAction.ConfirmCustodianCodes(custodianCodes, userEmailAddress);
+
+                        if (userConfirmation)
+                        {
+                            if (userStatus.Equals(UserStatus.Active))
+                            {
+                                adminAction.AddLas(custodianCodes, foundUserOrDefault);
+                            }
+                            else if (userStatus.Equals(UserStatus.New))
+                            {
+                                adminAction.TryCreateUser(userEmailAddress, custodianCodes);
+                            }
+                    }
+                    break;
+            }
+        }
+
+        private class DatabaseOperation : IDatabaseOperation
+        {
+            private readonly HerDbContext dbContext;
+            private readonly OutputProvider outputProvider;
+
+            public DatabaseOperation(HerDbContext dbContext, OutputProvider outputProvider)
+            {
+                this.dbContext = dbContext;
+                this.outputProvider = outputProvider;
+            }
+
+            public List<User> GetUsersWithLocalAuthorities()
+            {
+                return dbContext.Users
+                    .Include(user => user.LocalAuthorities)
+                    .ToList();
+            }
+
+            public void RemoveUserOrLogError(User? user)
+            {
+                using var dbContextTransaction = dbContext.Database.BeginTransaction();
+                try
+                {
+                    switch (user)
+                    {
+                        // removing a user also deletes all associated rows in the LocalAuthorityUser table
+                        case null:
+                            outputProvider.Output("User not found");
+                            break;
+                        default:
+                        {
+                            {
+                                dbContext.Users.Remove(user);
+                                dbContext.SaveChanges();
+                                var deletionConfirmation = outputProvider.Confirm(
+                                    "Attention! Deletion from a database. Are you sure you want to commit this transaction? (y/n)");
```

Comment:
Would be nice if this were more specific. Something like `This will delete user {username} and all associated rows from the database. Are you sure you want to commit this transaction?`

Reply:
Amended to be more specific :)